src/components/App.js

```diff
@@ -1,28 +1,60 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect } from "react";
+
+import { 
+    getProducts 
+} from "../api";
+
+import { 
+    Product, 
+    SingleProduct, 
+    Cart,
+    NavBar
+} from "./";
+
+import {
+  BrowserRouter as Router,
+  Route,
+  Switch,
+  Redirect,
+} from "react-router-dom";
 
 import {
-  getSomething
-} from '../api';
+    getLocalToken
+} from "../util";
 
 const App = () => {
-  const [message, setMessage] = useState('');
+    const [products, setProducts] = useState([]);
+    const [token, setToken] = useState('');
+    const [user, setUser] = useState({});
 
   useEffect(() => {
-    getSomething()
-      .then(response => {
-        setMessage(response.message);
-      })
-      .catch(error => {
-        setMessage(error.message);
-      });
-  });
+    getProducts().then(setProducts);
+      if (getLocalToken()) {
+          setToken(getLocalToken());
+      }
+  }, []);
 
   return (
-    <div className="App">
-      <h1>Hello, World!</h1>
-      <h2>{ message }</h2>
-    </div>
+    <>
+      <div className="App">
+        <NavBar 
+            token={token} 
+            setToken={setToken}
+            setUser={setUser}/>
+        <Route exact path="/cart">
+          < Cart />
+        </Route>
+        <Route exact path="/products">
+          {products.map((product) => {
+            return <Product key={product.id} product={product} />;
+          })}
+        </Route>
+        <Route exact path="/products/:productId">
+            <SingleProduct />
```

Comment:
Use the same component for Product and SingleProduct, and "toss out" the other 👍